Server/routes/route.js

```diff
@@ -1,46 +1,42 @@
+const { validateParcels, validateRegister, validateLogin, validateToken, validateSuperAdmin, validateAdmin, validateLocationUpdate } = middlewares;
+
+const { updateParcelDestination, cancelParcelOrder, registerUser, login, createAdmin  } = usersControllers;
+
+const { createNewParcel, getParcelsByUser, getAllParcels, getAllParcelsBySpecificUser, updateParcelStatus, updateParcelPresentLocation } = parcelController;
 
 const router = express.Router();
 
 // this is for the home route
 router.get('/', (req, res) => {
   res.status(200).send('welcome to SendIt');
 });
-
 // this is the route for creating parcels
-// CREATE PARCELS
-router.post('/parcels', validateParcels, validateToken, parcelController.createNewParcel);
+router.post('/parcels', validateParcels, validateToken, createNewParcel);
 // this is the route to get all parcels for a user
-// GET ALL PARCELS
-router.get('/parcels/user', parcelController.getParcelsByUser);
-// this is the route to get a specific parcel
-// GET A SPECIFIC PARCEL
-// router.get('/parcels/:id', parcelController.getSpecificParcel);
-// this is to change the dropofflocation of a parcel
-// PUT IN A NEW DROPOFFLOCATION
-router.put('/parcels/:pid/newlocation', validateToken, validateAdmin, parcelController.updateParcelPresentLocation);
+router.get('/parcels',validateToken, getParcelsByUser);
+// this is to enable a user change the dropoff location of a parcel
+router.patch('/parcels/:pid/destination',validateLocationUpdate, validateToken, updateParcelDestination);
 // this is to enable a user to cancel a parcel delivery order
-router.put('/parcels/:pid/cancel', validateToken, usersControllers.cancelParcelOrder);
-// this is to delete a specific parcel order
-// DELETE A PARCEL ORDER
-// router.delete('/parcels/:id', parcelController.deleteSpecificParcel);
-
+router.patch('/parcels/:pid/cancel', validateToken, cancelParcelOrder);
 // this is to register a new user
-// POST A NEW USER
-router.post('/auth/register', validateRegister, usersControllers.registerUser);
+router.post('/auth/register', validateRegister, registerUser);
 // this is to login in an existing user//
-router.post('/auth/login', validateLogin, usersControllers.login);
+router.post('/auth/login', validateLogin, login);
 // fetch all parcels in the application
 // ------------------------admin only------------------
 // GET ALL PARCELS IN THE APP (accessible to admin only)
-router.get('/parcels', validateToken, validateAdmin, parcelController.getAllParcels);
+router.get('/parcels/users', validateToken, validateAdmin, getAllParcels);
+// this is to get all parcels for a specific user
+router.get('/parcels/:uid', validateToken, validateAdmin, getAllParcelsBySpecificUser)
 // this is the route for an admin to update the status of a parcel delivery order
-router.put('/parcels/:pid/status', validateToken, validateAdmin, parcelController.updateParcelStatus);
+router.patch('/parcels/:pid/status', validateToken, validateAdmin, updateParcelStatus);
```

Comment:
this is supposed to be a `PUT` and not a `PATCH`. Are you sure you are looking at the rubrics?

Reply:
yes the rubics says patch

Reply:
![screenshot from 2018-11-22 17-50-31](https://user-images.githubusercontent.com/38977569/48915529-458e2180-ee7f-11e8-80bd-1e8fa640ed13.png)

Reply:
No it doesn't!!!
<img width="750" alt="screen shot 2018-11-22 at 5 51 10 pm" src="https://user-images.githubusercontent.com/8380409/48915606-96057f00-ee7f-11e8-9c12-0fa55029a7ee.png">

Reply:
it has been fixed